File: lc/length-of-last-word.py
"""
 a b 
   r
l
"""
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        n = len(s)
        r = n - 1
        while r >= 0 and s[r] == ' ':
            r -= 1
        l = r
        while l >= 0 and s[l] != ' ':
            l -= 1
        return r - l

File: lc/go/length-of-last-word.go
/*
remove spaces from back
then count from first char until space or start of string
" joy  "
    f
  l

"joy"
 l f

 "j"
  f
  l

"d a  "
   r
  l
*/
func lengthOfLastWord(s string) int {
    n := len(s)
    var l, r int
    for r = n - 1; r >= 0; r -- {
        if s[r] != ' ' {
            break
        }
    }
    for l = r; l >= 0; l -- {
        if s[l] == ' ' {
            break
        }
    } 
    return r - l 
}